src/infrasys/time_series_models.py

```diff
@@ -136,25 +136,24 @@ def aggregate(cls, ts_data: list[Self]) -> Self:
             raise InconsistentTimeseriesAggregation(msg)
 
         # Aggregate data
-        is_quantity = issubclass(next(iter(unique_props["data_type"])), BaseQuantity)
-        magnitude_type = (
-            type(ts_data[0].data.magnitude)
-            if is_quantity
-            else next(iter(unique_props["data_type"]))
-        )
+        data_type = next(iter(unique_props["data_type"]))
+        is_quantity = issubclass(data_type, BaseQuantity)
+        magnitude_type = type(ts_data[0].data.magnitude) if is_quantity else data_type
 
         # Aggregate data based on magnitude type
-        if issubclass(magnitude_type, pa.Array):
-            new_data = sum(
+        if issubclass(magnitude_type, pa.Array) and is_quantity:
+            new_data = agg_func(
```

Comment:
I think we could simply this cases. In all
the instances you have array and if it is pa.array we just need to convert it to numpy.